src/core/AutomatableModel.cpp

```diff
@@ -132,32 +133,37 @@ void AutomatableModel::saveSettings( QDomDocument& doc, QDomElement& element, co
 	}
 
 	if( m_controllerConnection && m_controllerConnection->getController()->type()
-				!= Controller::DummyController )
-	{
-		QDomElement controllerElement;
-
-		// get "connection" element (and create it if needed)
-		QDomNode node = element.namedItem( "connection" );
-		if( node.isElement() )
-		{
-			controllerElement = node.toElement();
-		}
-		else
-		{
-			controllerElement = doc.createElement( "connection" );
-			element.appendChild( controllerElement );
-		}
+				!= Controller::DummyController) {
+		// Skip saving MIDI connections if we're saving project and
+		// the discardMIDIConnections option is true.
+		if (!Engine::getSong()->isSavingProject()
+			|| !Engine::getSong()->getSaveOptions().discardMIDIConnections.value()
+			|| m_controllerConnection->getController()->type() != Controller::MidiController) {
```

Comment:
I think we can improve readability here. For example:
```cpp
auto controllerType = m_controllerConnection->getController()->type();
bool skipMidiController = Engine::getSong()->isSavingProject()
		&& Engine::getSong()->getSaveOptions().discardMIDIConnections.value();

if (m_controllerConnection && controllerType != Controller::DummyController
		&& !(skipMidiController&& controllerType == Controller::MidiController))
```
It will also reduce indentations. However, it's up to you.

Reply:
Thanks! I wasn't sure how to write it properly,